File: docs/examples/rs/examples/buckets_rename.rs
use reduct_rs::{ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a new client with the API URL and API token
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();

    // Get the bucket 'example-bucket'
    let bucket = client.get_bucket("example-bucket").await?;

    // Rename the entry 'entry_1' to 'entry_2'
    bucket.rename_entry("entry_1", "entry_2").await?;

    // Check that the entry was renamed
    let entries = bucket.entries().await?;
    let entry_names: Vec<String> = entries.iter().map(|entry| entry.name.clone()).collect();
    assert!(entry_names.contains(&"entry_2".to_string()));
    assert!(!entry_names.contains(&"entry_1".to_string()));

    Ok(())
}

File: versioned_docs/version-1.20.x/examples/py/src/buckets_rename.py
from reduct import Client, ReductError


async def rename_bucket():
    # Create a client with the base URL and API token
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:

        # Rename the bucket with the name "bucket-to-rename"
        bucket = await client.get_bucket("bucket-to-rename")
        await bucket.rename("bucket-renamed")

        # Check that the bucket was renamed
        assert await client.get_bucket("bucket-renamed")
        try:
            await client.get_bucket("bucket-to-rename")
        except ReductError as e:
            # The bucket should not exist anymore
            assert e.status_code == 404


if __name__ == "__main__":
    import asyncio

    asyncio.run(rename_bucket())
